src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -2,36 +2,62 @@ package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
 
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.util.UriComponentsBuilder
+import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthGroupID
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
 
-val AuthServiceProviderGroupPrefix = "INT_SP_"
+class UnverifiedEmailException : RuntimeException()
 
-data class AuthGroup(
+data class UserDetail(
+  val firstName: String,
+  val email: String,
+)
+
+private const val AuthServiceProviderGroupPrefix = "INT_SP_"
+
+private data class AuthGroupResponse(
   val groupCode: String,
   val groupName: String,
 )
 
+private data class AuthUserDetailResponse(
+  val firstName: String,
+  val email: String,
+  val verified: Boolean,
+)
+
+private data class UserEmailResponse(
+  val email: String,
+)
+
+private data class UserDetailResponse(
+  val name: String,
+)
+
 @Service
 class HMPPSAuthService(
-  @Value("\${hmppsauth.api.locations.user-groups}") private val userGroupsLocation: String,
+  @Value("\${hmppsauth.api.locations.auth-user-groups}") private val authUserGroupsLocation: String,
+  @Value("\${hmppsauth.api.locations.auth-user-detail}") private val authUserDetailLocation: String,
+  @Value("\${hmppsauth.api.locations.user-email}") private val userEmailLocation: String,
+  @Value("\${hmppsauth.api.locations.user-detail}") private val userDetailLocation: String,
```

Comment:
_not a blocker_

🙋‍♂️ Is there a need to have these locations as configuration? I'm assuming they won't change between environments, we don't need to use them elsewhere and if they change in _hmpps-auth_, we most likely to have other code changes to do as well.

There is probably an additional benefit with using the 2nd option: the reader does not have to look up a value from the config to understand what it's doing:

```kt
// 1
val url = UriComponentsBuilder.fromPath(authUserGroupsLocation)
  .buildAndExpand(user.userName)

// 2
val url = UriComponentsBuilder.fromPath("/api/authuser/{username}/groups")
  .buildAndExpand(user.userName)
```

What do you think?

Reply:
yeh i kind of agree. originally i liked having all the URLs in config because they kinda feel like config to me. but i agree that it's easier to read them in place and you're less likely to use the wrong one. 

can i put up a seperate PR to change all the configured 'locations' in this way (there are some for community api too)?

Reply:
actually - i'll just change it in this PR now

Reply:
so as i've gone through making this change i've identified a couple of positives to having these URLs in config.

1. they are reusable. in one instance the value was used in multiple places
2. allows injection of the url by test classes which makes using mock webservers more reliable in tests.

considering these - i'm going to leave this change as it is for now and we can consider these pros and cons separately.